### PR TITLE
enhance: add materialized view search info for compatibility

### PR DESCRIPTION
**Pull request by PwzXxm**
/kind improvement

### CODE REVIEW

File: include/knowhere/comp/materialized_view.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#include <charconv>
+#include <cstdint>
+#include <nlohmann/json.hpp>
+#include <unordered_map>
+
+namespace knowhere {
+struct MaterializedViewSearchInfo {
+    std::unordered_map<int64_t, uint64_t> field_id_to_touched_categories_cnt;
+    bool is_pure_and = true;
+    bool has_not = false;
```
Comment (zhengbuqian) [inline include/knowhere/comp/materialized_view.h:23]: what does those mean? pls add comment

File: include/knowhere/config.h [hunk 2]
```diff
@@ -182,6 +187,29 @@ struct Entry<CFG_BOOL> {
...
+
+    Entry<CFG_MATERIALIZED_VIEW_SEARCH_INFO_TYPE>() {
+        val = nullptr;
+        default_val = std::nullopt;
+        type = 0x0;
+        desc = std::nullopt;
+    }
+
+    CFG_MATERIALIZED_VIEW_SEARCH_INFO_TYPE* val;
+    std::optional<MaterializedViewSearchInfo> default_val;
```
Comment (zhengbuqian) [inline include/knowhere/config.h:207]: use CFG_MATERIALIZED_VIEW_SEARCH_INFO_TYPE::value_type?

**Review by zhengbuqian — COMMENTED** [review]
the code mostly looks good but I don't have sufficient context, can you link a issue/doc?

**Comment by PwzXxm** [comment]
Mainly for https://github.com/milvus-io/milvus/issues/29892

File: include/knowhere/comp/materialized_view.h (hunk 1 above)
Comment (PwzXxm) [reply include/knowhere/comp/materialized_view.h:35]: My bad. Added.

**Pull request merged**